function-invokers/java-function-invoker/src/test/java/io/sk8s/invoker/java/function/FunctionConfigurationTests.java

```diff
@@ -31,52 +34,61 @@
 import org.springframework.test.context.TestPropertySource;
 import org.springframework.test.context.junit4.SpringRunner;
 
-import io.sk8s.invoker.java.function.FunctionConfiguration;
-
 import static org.hamcrest.Matchers.containsString;
 import static org.hamcrest.Matchers.is;
 import static org.junit.Assert.assertThat;
 
 @RunWith(SpringRunner.class)
-@SpringBootTest(classes = FunctionConfiguration.class)
+@SpringBootTest(classes = { FunctionConfiguration.class, FunctionProperties.class })
 public abstract class FunctionConfigurationTests {
 
-	@TestPropertySource(properties = { "function.jarLocation=file:target/test-classes",
-			"function.className=" +
+	@TestPropertySource(properties = { "function.uri=file:target/test-classes?io.sk8s.invoker.java.function.FunctionConfigurationTests.Doubler"
+	})
+	public static class SingleFunctionTests extends FunctionConfigurationTests {
+	
+		@Autowired
+		@Qualifier("function0")
+		private Function<Integer, Integer> function;
+	
+		@Test
+		public void testDouble() {
+			assertThat(function.apply(2), is(4));
+		}
+	}
+	
+	@TestPropertySource(properties = { "function.uri=file:target/test-classes?" +
 					"io.sk8s.invoker.java.function.FunctionConfigurationTests.NumberEmitter," +
 					"io.sk8s.invoker.java.function.FunctionConfigurationTests.Frenchizer"
 	})
 	public static class SupplierCompositionTests extends FunctionConfigurationTests {
 
 		@Autowired
-		@Qualifier("function")
-		private Supplier<String> function;
+		@Qualifier("function0")
+		private Supplier<Integer> function;
 
 		@Test
 		public void testComposition() {
-			assertThat(function.get(), is("un"));
+			assertThat(function.get(), is(1));
 		}
 	}
 
-	@TestPropertySource(properties = { "function.jarLocation=file:target/test-classes",
-			"function.className=" +
+	@TestPropertySource(properties = { "function.uri=file:target/test-classes?" +
 					"io.sk8s.invoker.java.function.FunctionConfigurationTests.Doubler," +
 					"io.sk8s.invoker.java.function.FunctionConfigurationTests.Frenchizer"
 	})
 	public static class FunctionCompositionTests extends FunctionConfigurationTests {
 
 		@Autowired
-		@Qualifier("function")
+		@Qualifier("function1")
 		private Function<Integer, String> function;
 
 		@Test
 		public void testComposition() {
-			assertThat(function.apply(2), is("quatre"));
```

Comment:
Related to my comment above I guess. This definitely does not test the same thing as before.
I guess composition is only supported while hitting this via http, so most of this test code can be removed?

Reply:
Agree that this is not testing the same thing. You could pull a composed function out of the `FunctionCatalog` (which isn't even in the context of these tests) and test the same thing that way, but then you'd only be testing features of Spring Cloud Function. I just wanted a sanity check that the functions had been created bu `FunctionConfiguration`.

Reply:
Same as above, this only tests the second function (function1) and hence is not testing composition